### DISCUSSION TITLE
多进程模式下，动态定时任务的执行问题

### DISCUSSION THREAD

**Discussion by svenjia** [eggjs/egg#5309] [Q&A]
请问在多进程模式下，在app.js中注册的动态定时任务，如果该worker挂掉后，定时任务还会不会执行？如果多个worker同时执行，那我业务逻辑，比如写入一定的数据到mysql，会不会写入多份

**Comment by fengmk2** [comment]
https://www.eggjs.org/zh-CN/basics/schedule#%E7%B1%BB%E5%9E%8B 选择 worker 模式，只会随机挑选一个 worker 执行。

**Comment by svenjia** [comment]
是这样的，我不是用egg自带的schedule。我是从数据库里读出来的数据，每条数据有执行相应任务的时间戳，我用node-schedual根据时间戳来注册的定时任务。所有worker应该都会执行，这样就容易造成同一任务执行多次  @fengmk2

**Reply by fengmk2** [reply]
这个就不是我能回答的了，node-schedual 没有使用过。

**Reply by svenjia** [reply]
谢谢，我去看下egg里面schedule是怎么实现随机worker执行的。
